tests/AsyncStreamTests.cs

```diff
@@ -137,10 +114,64 @@ public async Task GetAsyncEnumerator_With_Airports_Data()
                 Icao = "TX98", Iata = null, Name = "Hawkins Private Airport", City = "Godley",
                 State = "Texas", Country = CountryCode.US, Elevation = 975,
                 Coordinates = new(32.4751014709, -97.5009002686), TimeZone = "America/Chicago"
-            },
-        };
+            });
+
+    [Fact]
+    public async Task GetAsyncEnumerator_With_Airports_Data()
+    {
+        Airport? lastGoodAirport = null;
+        var actual = new List<Airport>();
+
+        try
+        {
+            var count = 0;
+
+            await using var stream = File.OpenRead(Path.Join("..", "..", "..", "data", "airports", "airports-array.json"));
+            await using var enumerator = AirportReader.GetAsyncEnumerator(stream, initialBufferSize: 10);
+            while (await enumerator.MoveNextAsync())
+            {
+                lastGoodAirport = enumerator.Current;
+                if (count++ % 5_000 is 0)
+                    actual.Add(lastGoodAirport);
+            }
+        }
+        catch (JsonException) when (lastGoodAirport is not null)
+        {
+            WriteLine($"Last Good Airport = {lastGoodAirport}");
+            throw;
+        }
+
+        Assert.Equal(ExpectedAirports, actual);
+    }
+
```

Comment:
Do we want to add tests that cover the error cases ("outer" and "inner") as well?

Reply:
Thanks for the clarification. Yes, those should be added, but since they're not there for `GetAsyncEnumerator` either at the moment, I'd say the added test is on par with the present coverage. Tests for both enumerator can be added in a future PR to increase coverage.